File: LeetCode/medium/Container_With_Most_Water.cxx
class Solution {
public:
    int maxArea(vector<int>& height) {
        int res_max = 0;
        int a = 0, b = height.size() - 1;
        
        while (a < b)
        {
            int v = (b - a) * std::min(height[a], height[b]);
            if (v > res_max)
                res_max = v;
            if (height[a] < height[b])
                ++a;
            else
                --b;
        }
        return res_max;
    }
};

File: LeetCode/medium/Container_With_Most_Water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        res = 0
        l = 0
        r = len(height)-1
        while l < r:
            v = min(height[l], height[r]) * (r-l)
            res = max(res, v)
            if height[l] < height[r]:
                l += 1
            else:
                r -= 1
        return res
